drivers/mpu9150/mpu9150.c

```diff
@@ -0,0 +1,610 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     mpu9150
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the MPU-9150 9-Axis Motion Sensor
+ *
+ * @author      Fabian Nack <nack@inf.fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "mpu9150.h"
+#include "mpu9150-regs.h"
+#include "periph/i2c.h"
+#include "vtimer.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define REG_RESET           (0x00)
+#define MAX_VALUE           (0x7FFF)
+
+/* Default config settings */
+static const mpu9150_status_t DEFAULT_STATUS = {
+    .accel_pwr = PWR_ON,
+    .gyro_pwr = PWR_ON,
+    .compass_pwr = PWR_ON,
+    .gyro_fsr = GYRO_FSR_250DPS,
+    .accel_fsr = ACCEL_FSR_16G,
+    .sample_rate = 0,
+    .compass_sample_rate = 0,
+    .compass_x_adj = 0,
+    .compass_y_adj = 0,
+    .compass_z_adj = 0,
+};
+
+/* Internal function prototypes */
+static int compass_init(mpu9150_t *dev);
+static void conf_bypass(mpu9150_t *dev, uint8_t bypass_enable);
+static void conf_lpf(mpu9150_t *dev, uint16_t rate);
+
+/*---------------------------------------------------------------------------*
+ *                          MPU9150 Core API                                 *
+ *---------------------------------------------------------------------------*/
+
+int mpu9150_init(mpu9150_t *dev, i2c_t i2c, mpu9150_hw_addr_t hw_addr,
```

Comment:
This signature differs from definition [above](https://github.com/RIOT-OS/RIOT/pull/2368/files#diff-6d056d884f6ca8fe838f30f273e1c2dfR183). I think it's better to use `uint8_t` instead of `mpu9150_hw_addr_t` and `mpu9150_comp_addr_t`.